var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -0,0 +1,89 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Warpx(MakefilePackage):
+    """WarpX is an advanced electromagnetic Particle-In-Cell code. It supports
+    many features including Perfectly-Matched Layers (PML) and mesh refinement.
+    In addition, WarpX is a highly-parallel and highly-optimized code and
+    features hybrid OpenMP/MPI parallelization, advanced vectorization
+    techniques and load balancing capabilities.
+    """
+
+    homepage = "https://ecp-warpx.github.io/index.html"
+    url      = "https://bitbucket.org/berkeleylab/warpx.git"
+
+    version('master', git='https://bitbucket.org/berkeleylab/warpx.git', tag='master')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('gcc')
```

Comment:
Does WarpX really depend on GCC (as a dependency) or is this the compiler known to work? In Spack, compilers are part of the specification (spec) with `%compiler` and don't need to be listed as dependencies. One can list conflicts of known, not working compilers though.

Refs:
https://spack.readthedocs.io/en/latest/basic_usage.html#specs-dependencies
https://spack.readthedocs.io/en/latest/packaging_guide.html#conflicts

Reply:
This was there to force WarpX to be built with gcc. This is not necessary and was only there to get started. I'll fix this (using coding similar to what is done in picsar).